src/ixdat/techniques/ms.py

```diff
@@ -66,100 +36,40 @@ def __init__(self, name, **kwargs):
 
     @property
     def ms_calibration(self):
-        ms_cal_list = []
-        tspan_bg = None
-        signal_bgs = {}
-        for cal in self.calibration_list:
-            ms_cal_list = ms_cal_list + getattr(cal, "ms_cal_list", [])
-            for mass, bg in getattr(cal, "signal_bgs", {}).items():
-                if mass not in signal_bgs:
-                    signal_bgs[mass] = bg
-            tspan_bg = tspan_bg or getattr(cal, "tspan_bg", None)
-        return MSCalibration(ms_cal_results=ms_cal_list, signal_bgs=signal_bgs)
+        return self.calculators["MS calibration"]
+
+    @property
+    def siq_calculator(self):
+        return self.calculators["siq calculator"]
 
     @property
     def signal_bgs(self):
-        return self.ms_calibration.signal_bgs
+        return self.calculators["MS background"]
 
-    def set_bg(self, tspan_bg=None, mass_list=None):
+    def set_bg(self, tspan=None, tspan_bg=None, mass_list=None):
         """Set background values for mass_list to the average signal during tspan_bg."""
-        mass_list = mass_list or self.mass_list
-        tspan_bg = tspan_bg or self.tspan_bg
-        signal_bgs = {}
-        for mass in mass_list:
-            t, v = self.grab(mass, tspan_bg)
-            signal_bgs[mass] = np.mean(v)
-        self.add_calibration(MSCalibration(signal_bgs=signal_bgs))
+        tspan = tspan or tspan_bg
+        background = MSBackgroundSet.from_measurement_point(
+            measurement=self, tspan=tspan, mass_list=mass_list
+        )
+        self.add_calculator(background)
 
     def reset_bg(self, mass_list=None):
-        """Reset background values for the masses in mass_list"""
-        mass_list = mass_list or self.mass_list
-        new_signal_bgs = {}
-        for mass in mass_list:
-            if mass in self.signal_bgs:
-                new_signal_bgs[mass] = 0
-        self.add_calibration(MSCalibration(signal_bgs=new_signal_bgs))
-
-    def grab(
```

Comment:
Do you have some deprecation of `MSMeasurement.grab()` left in there somewhere as well? Didnt see it at quick glance.

Reply:
No, `Meausrement.grab()` is still there!
`MSMeasurement.grab()` doesn't add anything anymore, since the `remove_background` argument that was in `MSMeasurement.grab()` got bumped up to `Measurement.grab()`, and there's nothing special about grabbing fluxes now that `siqCalculator` is a normal `Calculator`.